definitions/tools/mutect.cwl

```diff
@@ -2,75 +2,66 @@
 
 cwlVersion: v1.0
 class: CommandLineTool
-label: "mutect2 (GATK 3.6)"
-baseCommand: ["/usr/bin/java", "-jar", "/opt/GenomeAnalysisTK.jar", "-T", "MuTect2"]
+label: "Mutect2 (GATK 4)"
+
+baseCommand: ["/bin/bash", "Mutect2.sh"]
+
 requirements:
     - class: ResourceRequirement
       ramMin: 20000
       tmpdirMin: 100000
     - class: DockerRequirement
-      dockerPull: "mgibio/cle:v1.3.1"
+      dockerPull: "broadinstitute/gatk:4.1.2.0"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'Mutect2.sh'
+        entry: |
+            set -o pipefail
+            set -o errexit
+
+            /gatk/gatk Mutect2 -O $1 -R $2 -I $3 -tumor $5 -I $4 -normal $6 -L $7
+            gunzip mutect.vcf.gz
+            sed "s/\t$5\t/\tTUMOR\t/g; s/\t$6/\tNORMAL/g" mutect.vcf > mutect_final.vcf
+            bgzip mutect_final.vcf
+            tabix mutect_final.vcf.gz
```

Comment:
You still need to add the steps we talked about, that will only apply the sample_name change to the appropriate header line.  Something like:

    grep "^##" mutect.vcf >mutect.final.vcf
    grep "^#CHROM" mutect.vcf | sed "s/\t$5\t/\tTUMOR\t/g; s/\t$6/\tNORMAL/g" >> mutect_final.vcf
    grep -v "^#" mutect.vcf >> mutect_final.vcf

Reply:
We should also think about whether it's reasonable to future-proof this to a time when we actually do use the sample names throughout, instead of "TUMOR" and "NORMAL".